lua_modules/http/httpserver.lua

```diff
@@ -23,54 +23,49 @@ do
   -- response methods
   ------------------------------------------------------------------------------
   local make_res = function(csend, cfini)
-   local send = function(self, data, status)
-    -- TODO: req.send should take care of response headers!
-    if self.send_header then
-      csend("HTTP/1.1 ")
-      csend(tostring(status or 200))
-      -- TODO: real HTTP status code/name table
-      csend(" OK\r\n")
-      -- we use chunked transfer encoding, to not deal with Content-Length:
-      --   response header
-      self:send_header("Transfer-Encoding", "chunked")
-      -- TODO: send standard response headers, such as Server:, Date:
+   -- send("hello world\r\n") will send the data
+    -- send() will close connection
+    local send = function(data)
+      if data then
+        -- chunked transfer encoding
+        csend(("%X\r\n"):format(#data))
+        csend(data)
+        csend("\r\n")
+      else
+        -- finalize chunked transfer encoding
+        csend("0\r\n\r\n")
+        -- close connection
+        cfini()
+      end
     end
-    if data then
-      -- NB: no headers allowed after response body started
-      if self.send_header then
-        self.send_header = nil
-        -- end response headers
+    local send_header = function(header)
+      if type(header) ~= "table" then
+        print("httpserver.lua: header need a table! eg: {status = 200, headers = {Connection = \"close\"}}")
```

Comment:
Please throw an error here.  This is not the kind of mistake that should just merit printing.

If you are unfamiliar with the "kinds" of errors I mean, consider the taxonomy given in http://www.cs.cmu.edu/~410/lectures/L08a_Errors.pdf

Reply:
if res has 3 function. Throw error will needless